packages/plugins/apollo-managed-federation/src/index.ts

```diff
@@ -14,25 +14,25 @@ export function useManagedFederation(options: ManagedFederationPluginOptions = {
   // Start as soon as possible to minimize the wait time of the first schema loading
   supergraphManager.start();
 
-  // Wait for the first schema to be loaded before before allowing requests to be parsing
-  // We can then remove the onRequestParse hook to avoid async cost on every request
-  const waitForInitialization = new Promise(resolve => {
-    supergraphManager.on('schema', () => {
-      plugin.onRequestParse = undefined;
-      resolve(null);
-    });
-  });
-
   const plugin: Plugin = {
     onPluginInit({ setSchema }) {
       if (supergraphManager.schema) {
         setSchema(supergraphManager.schema);
+      } else {
+        // Wait for the first schema to be loaded before before allowing requests to be parsed
+        // We can then remove the onRequestParse hook to avoid async cost on every request
+        const waitForInitialization = new Promise<void>(resolve => {
+          supergraphManager.on('schema', () => {
+            plugin.onRequestParse = undefined;
+            resolve();
+          });
+        });
+        plugin.onRequestParse = async () => {
+          await waitForInitialization;
```

Comment:
```suggestion
          return waitForInitialization;
```
Why don't we just return?

Reply:
I found it more expressive this way, but I can change if you want. Not sure if it changes something to performance ^^'

Reply:
You know `async` creates a new Promise instance. But if we return the promise as-is, it can be just handled directly.

Reply:
Oh you call it once anyway. So it is not a big deal. Ok ok...